packages/account-abstraction/src/SmartAccountAPI.ts

```diff
@@ -182,21 +184,23 @@ export class SmartAccountAPI extends BaseAccountAPI {
    * - if gas or nonce are missing, read them from the chain (note that we can't fill gaslimit before the wallet is created)
    * @param info
    */
-  async createUnsignedUserOp(info: TransactionDetailsForBatchUserOp): Promise<UserOperation> {
+  async createUnsignedUserOp(info: TransactionDetailsForBatchUserOp, vGasLimit?: BigNumber): Promise<UserOperation> {
     const { callData, callGasLimit } = await this.encodeUserOpCallDataAndGasLimit(info)
     console.log(callData, callGasLimit)
 
     const initCode = await this.getInitCode()
 
-    const verificationGasLimit = BigNumber.from(await this.getVerificationGasLimit())
+    let verificationGasLimit = vGasLimit
+    if ( !vGasLimit )
+    verificationGasLimit = BigNumber.from(await this.getVerificationGasLimit())
 
     let { maxFeePerGas, maxPriorityFeePerGas } = info
     // Note: Custom should be equal if it's for non EIP1559
+    const chainId = this.clientConfig.chainId
 
     if (maxFeePerGas == null || maxPriorityFeePerGas == null) {
-      const feeData = await this.provider.getFeeData()
-      Logger.log('EIP1559 feeData', feeData)
-      const chainId = this.clientConfig.chainId
+      const feeData = await this.httpRpcClient.getUserOpGasPrices(chainId)
```

Comment:
let's have proper error handling here, if RPC url call to eth_getUserOpGasPrices then it should log and fallback to getting these values from provider like we're doing now.